waspc/src/Wasp/Generator/Common.hs

```diff
@@ -11,15 +13,36 @@ import Text.Printf (printf)
 -- | Directory where the whole web app project (client, server, ...) is generated.
 data ProjectRootDir
 
+type SemanticVersion = (Int, Int, Int)
+
+compatibleWith :: String
+compatibleWith = "^"
+
+getMajor :: SemanticVersion -> Int
+getMajor (major, _, _) = major
+
 -- | Node version that node packages generated by this generator expect.
 --   (major, minor, patch)
-nodeVersion :: (Int, Int, Int)
-nodeVersion = (12, 18, 0) -- Latest LTS version.
+nodeVersion :: SemanticVersion
+nodeVersion = (16, 0, 0) -- Latest LTS version.
+
+nodeSemverString :: String
+nodeSemverString = makeSemverString compatibleWith nodeVersion
+
+npmVersion :: SemanticVersion
+npmVersion = (8, 0, 0) -- Latest LTS version.
+
+npmSemverString :: String
+npmSemverString = makeSemverString compatibleWith npmVersion
+
+prismaVersion :: SemanticVersion
+prismaVersion = (3, 9, 1)
+
+prismaSemverString :: String
+prismaSemverString = makeSemverString "" prismaVersion
```

Comment:
The prefix is empty because I decided to pin the _prisma_ version down to exactly `3.9.1` (without tolerance) to avoid any possible compatibility issues. We can consider updating it in a different PR (I believe version 3.10 is out).

Reply:
I think that is a good idea 👍🏻

Reply:
Maybe just stick with `prismaVersionAsText` -> then it is clear it is not a semverString/VersionBoundary, but instead it is specific version. And then later, when I want to use it, I know what I am using, I know that I am using a specific version, because that is a decision I am making at that call site, not here (hm is it? Ok, maybe this is a bit arguable. but still I think being more precise can't hurt).

Reply:
Hm, not sure about this one.

The name matches the other ones because this variable gets embedded into the template in a way that also supports specifying version bounds (and it is a version bound, albeit a strict one). I don't think the fact that the this particular version bound happen to cover only a single version (which semver communicates by not specifying a prefix) should affect the variable's name.

If we want the string `"3.9.1"` specifically (i.e., a version string with no extra semver stuff inside it), I think we should call `show version` or something similar. In other words, string representations of versions should not mix with bound representation strings.

Reply:
Got it! I wanted to suggest that we have only prisma version and then use that everywhere, so it is explicit we are always using a specific version, not a version bound that by accident turns out to be very specific. Because I don't think we will ever want to relax it. And knowing that it is a very specific version gives us a bit more knowledge in the places where we use it.

But we can also go the way you describe, call this version bound and use it all around. It is not a big problem, we can always refactor if we figure out better approach.

TLDR: I argumented a bit but don't have super strong opinion, choose what you like better and I am ok with it!